### PR TITLE
Update Ballerina Syntax Guide to include missing variables

### PR DESCRIPTION
**Pull request by RukshiW**
## Purpose
Fixes https://github.com/wso2-enterprise/choreo/issues/6747

### CODE REVIEW

File: en/docs/references/ballerina-expression-syntax.md [hunk 1]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
   +950.930
   ```
 
+### decimal
+
+The `decimal` type supports integer values with up to 34 decimal points, and supports the same arithmetic operators as `int`. When entering literals, you can use `d` as the suffix.
```
Comment (Nashaath) [inline en/docs/references/ballerina-expression-syntax.md:42]:
```suggestion
The `decimal` type supports integer values with up to 34 decimal points and supports the same arithmetic operators as `int`. When entering literals, you can use `d` as the suffix.
```

File: en/docs/references/ballerina-expression-syntax.md [hunk 2]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
...
+
+This type does not support `infinity`, `NaN` or negative zero.
```
Comment (Nashaath) [inline en/docs/references/ballerina-expression-syntax.md:44]:
```suggestion
This type does not support `infinity`, `NaN`, or negative zero.
```

File: en/docs/references/ballerina-expression-syntax.md [hunk 3]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
...
+
+The `decimal` type is a separate basic type, and therefore counts as `anydata`.
```
Comment (Nashaath) [inline en/docs/references/ballerina-expression-syntax.md:46]:
```suggestion
The `decimal` type is a separate basic type and therefore counts as `anydata`.
```

File: en/docs/references/ballerina-expression-syntax.md [hunk 4]
```diff
@@ -68,6 +88,50 @@ To initialize a variable of type `var` in Choreo, you can use one of the followi
...
+
+To initialize a variable of type `error` in Choreo, you can enter an expression as follows:
+
+```ballerina
+error("The item requested is not found.")
+```
+
+### any
+
+The any type can represent a variable of any type except the error type. When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type any. Values of these variables can come from dynamic content such as the request and response message references and reference variable types.
```
Comment (Nashaath) [inline en/docs/references/ballerina-expression-syntax.md:105]: Code formatting needs to be applied for the term any:
"The `any` type can represent. . . ."

File: en/docs/references/ballerina-expression-syntax.md [hunk 5]
```diff
@@ -68,6 +88,50 @@ To initialize a variable of type `var` in Choreo, you can use one of the followi
...
+  
+        123
+  
+        349.76
+    ```
+    
+
+### anydata
+
+The `anydata` type consists of pure values whose basic type is not `error`.
```
Comment (Nashaath) [inline en/docs/references/ballerina-expression-syntax.md:123]: Suggestion:
Should we consider replacing "whose" with "where the".

Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md:123]: How about "of which the"?

File: en/docs/references/ballerina-expression-syntax.md (hunk 4 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Done

File: en/docs/references/ballerina-expression-syntax.md (hunk 5 above)
Comment (Nashaath) [reply en/docs/references/ballerina-expression-syntax.md]: +1

File: en/docs/references/ballerina-expression-syntax.md [hunk 6]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
   +950.930
   ```
 
+### decimal
+
+The `decimal` type supports integer values with up to 34 decimal points and supports the same arithmetic operators as `int`. When entering literals, you can use `d` as the suffix.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:42]: IMO, it's more useful to say something like (quoting from the spec)

```suggestion
The `decimal` type corresponds to a subset of IEEE 754-2008 128-bit decimal (radix 10) floating point numbers.

It supports the same arithmetic operations as `int` and `float`. When entering literals, you can use `d` as the suffix can be used to indicate that the value is a `decimal` value.
```

File: en/docs/references/ballerina-expression-syntax.md [hunk 7]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
...
+
+This type does not support `infinity`, `NaN`, or negative zero.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:44]: Why are we saying negative zero is not supported?

Comment (MaryamZi) [reply en/docs/references/ballerina-expression-syntax.md:44]: Spec says "NaN, infinities and **subnormals** are not supported".

File: en/docs/references/ballerina-expression-syntax.md [hunk 8]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
...
+
+The `decimal` type is a separate basic type and therefore counts as `anydata`.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:46]: The fact that `decimal` is a separate basic type doesn't affect it being a subtype of `anydata`.

```suggestion
The `decimal` type is a separate basic type and is a subtype of `anydata`.
```

File: en/docs/references/ballerina-expression-syntax.md [hunk 9]
```diff
@@ -37,6 +37,26 @@ To initialize a variable of type `float` in Choreo, you can use one of the follo
...
+###### Examples:
+
+To initialize a variable of type `decimal` in Choreo, you can use one of the following expressions:
+
+  ```ballerina
+  0.33536363363636391745284629571635428
+  
+  -8593.9926666666666666663333333333
+  
+  +950.930999999992727272777777777774
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:57]: Shall we add an example with the `d` suffix also?

File: en/docs/references/ballerina-expression-syntax.md [hunk 10]
```diff
@@ -68,6 +88,50 @@ To initialize a variable of type `var` in Choreo, you can use one of the followi
 "Hello!"
 ```
 
+### error
+
+The `error` type supports error values that include a string message. The expressions to initialize this variable type needs to be entered in the `error("<ERROR_STRING>")` format.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:93]:
```suggestion
The `error` type represents error values. Error values contain a message of type `string` and optionally a `cause` and additional details about the error. The expressions to initialize a variable of type `error` needs to be entered in the `error("<MESSAGE>")` format.
```

Shouldn't we also include examples of passing the cause and/or details? Aren't these supported yet?

File: en/docs/references/ballerina-expression-syntax.md [hunk 11]
```diff
@@ -68,6 +88,50 @@ To initialize a variable of type `var` in Choreo, you can use one of the followi
...
+
+To initialize a variable of type `error` in Choreo, you can enter an expression as follows:
+
+```ballerina
+error("The item requested is not found.")
+```
+
+### any
+
+The `any` type can represent a variable of any type except the error type. When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type any. Values of these variables can come from dynamic content such as the request and response message references and reference variable types.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:105]:
```suggestion
The `any` type represents any value that is not an error. When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type `any`. Values assigned to a variable of type `any` can come from dynamic content such as the request and response message references and reference variable types.
```

Comment (MaryamZi) [reply en/docs/references/ballerina-expression-syntax.md:105]: Do we need to say "When you do not have prior knowledge of the type of a variable, you can assign it to a variable of type `any`"? This isn't quite correct IMO. We will always know the static type.

File: en/docs/references/ballerina-expression-syntax.md [hunk 12]
```diff
@@ -68,6 +88,50 @@ To initialize a variable of type `var` in Choreo, you can use one of the followi
...
+  
+        123
+  
+        349.76
+    ```
+    
+
+### anydata
+
+The `anydata` type consists of pure values of which the basic type is not `error`.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:123]: Shouldn't we also specify the exact types?

```
  () | boolean | int | float | decimal
    | string | xml
    | anydata[] | map<anydata> | table<map<anydata>>
```

File: en/docs/references/ballerina-expression-syntax.md [hunk 13]
```diff
@@ -68,6 +88,50 @@ To initialize a variable of type `var` in Choreo, you can use one of the followi
...
+
+###### Examples:
+
+To initialize a variable of type `anydata` in Choreo, you can enter expressions as follows:
+
+    ```ballerina
+        x2 = x1.clone();
+        
+        eq = (x1 == x2);
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:132]: Don't we usually mention only the right hand side expression?

File: en/docs/references/ballerina-expression-syntax.md [hunk 14]
```diff
@@ -492,3 +556,31 @@ To convert the `theRevenantJson` variable of type `json` to the record `Movie`,
 ```ballerina
 check theRevenantJson.cloneWithType(Movie);
 ```
+
+## XML
+
+XML is a textual format for representing a string or an XML object. The following topics describe these values.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:562]: We don't use the term XML object. Shall we revisit this entire section?

Relevant section in the spec - https://ballerina.io/ballerina-spec/spec.html#XML.

File: en/docs/references/ballerina-expression-syntax.md [hunk 15]
```diff
@@ -492,3 +556,31 @@ To convert the `theRevenantJson` variable of type `json` to the record `Movie`,
...
+
+### String values in XML
+
+Ballerina allows you to cast a string value in XML via variables of type `xml`.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:566]: This is also not correct.

File: en/docs/references/ballerina-expression-syntax.md [hunk 16]
```diff
@@ -492,3 +556,31 @@ To convert the `theRevenantJson` variable of type `json` to the record `Movie`,
...
+
+To initialize a variable of type `xml`, you can enter an expression as follows:
+
+```ballerina
+    xml `"Dear customer,"`
+```
+
+### XML objects
+
+You can define an XML object to hold XML values.
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:578]: I think we have to use the proper XML terms here.

File: en/docs/references/ballerina-expression-syntax.md [hunk 17]
```diff
@@ -492,3 +556,31 @@ To convert the `theRevenantJson` variable of type `json` to the record `Movie`,
...
+
+###### Examples:
+
+To declare a JSON object, you can enter an expression as follows:
```
Comment (MaryamZi) [inline en/docs/references/ballerina-expression-syntax.md:582]: The example is of an XML element.

File: en/docs/references/ballerina-expression-syntax.md (hunk 7 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md:46]:
```suggestion
This type does not support `infinity`, `NaN`, or subnormals.
```

File: en/docs/references/ballerina-expression-syntax.md (hunk 9 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Done

File: en/docs/references/ballerina-expression-syntax.md (hunk 13 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Corrected

File: en/docs/references/ballerina-expression-syntax.md (hunk 14 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Done

File: en/docs/references/ballerina-expression-syntax.md (hunk 15 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Rewrote this section.

File: en/docs/references/ballerina-expression-syntax.md (hunk 16 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Rewrote this section.

File: en/docs/references/ballerina-expression-syntax.md (hunk 17 above)
Comment (RukshiW) [reply en/docs/references/ballerina-expression-syntax.md]: Corrected

**Comment by RukshiW** [comment]
Obsolete

**Pull request closed without merging**